src/rpc/mod.rs

```diff
@@ -110,6 +113,41 @@ where
             expected_payment,
         )
     }
+
+    fn check_gas_limit(
+        &self,
+        parent_hash: &B256,
+        registered_gas_limit: u64,
+        block_gas_limit: u64,
+    ) -> RpcResult<()> {
+        let parent =
+            self.provider()
+                .header(parent_hash)
+                .to_rpc_result()?
+                .ok_or(internal_rpc_err(format!(
+                    "Parent block with hash {} not found",
+                    parent_hash
+                )))?;
+        if registered_gas_limit == 0
+            && block_gas_limit == calc_gas_limit(parent.gas_limit, 30_000_000)
+        {
+            // Prysm has a bug where it registers validators with a desired gas limit
+            // of 0. Some builders treat these as desiring gas limit 30_000_000. As a
+            // workaround, whenever the desired gas limit is 0, we accept both the
+            // limit as calculated with a desired limit of 0, and builders which fall
+            // back to calculating with the default 30_000_000.
+            return Ok(());
+        } else {
```

Comment:
I'd prefer unwrapping the else branch leaving it as plain if+early-return. I was forced to use one to keep the workaround minimal but since you neatly put this code in its own fn (👌 ) it's no longer needed. Less code better.

If you prefer this variant, np, just leave it like this.

Reply:
I agree, I actually had that version first but then reverted to this one to make it easier to compare against the existing implementation.

Will refactor though to get rid of the unnecessary else block.

Reply:
Interesting that clippy didn't complain about this🤔

Reply:
Adjusted according to your suggestion in: https://github.com/ultrasoundmoney/reth-payload-validator/pull/18/commits/06496cde32dc72563861f6b6a442ef8e5af5135a

Reply:
Note that my change removed the comment. Lmk if you want me to add it back in